### PR TITLE
Add scripts to perform generic tests

### PR DESCRIPTION
**Pull request by aragua**
Currently tests are related to boards but some of them are generic and can be developed in a common script.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,6 +10,7 @@ HELPERS := assert_device_present \
 	   assert_partition_found \
 	   assert_sysfs_attr_present \
 	   bootrr \
+	   bootrr-auto \
```
Comment (ndechesne) [inline Makefile:14]: something similar is available in https://github.com/andersson/bootrr/pull/19/. we need one or the other..
@alimon  can you please review this one?

File: helpers/assert_file_is_empty [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+TEST_CASE_ID="$1"
+FILEPATH="$2"
+TIMEOUT="${4:-1}"
+
+if [ -z "${TEST_CASE_ID}" -o -z "${FILEPATH}" ]; then
+	echo "Usage: $0 <test-case-id> <file path> [<timeout>]"
+	exit 1
+fi
+
+timeout ${TIMEOUT} [ -s ${FILEPATH} ] || test_report_exit pass
```
Comment (ndechesne) [inline helpers/assert_file_is_empty]: probably better with double quotes around ${FILEPATH}? it looks like we should do that in other places..

File: helpers/bootrr-generic-tests [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+assert_file_is_empty deferred-probe-empty /sys/kernel/debug/devices_deferred
```
Comment (ndechesne) [inline helpers/bootrr-generic-tests:3]: is that clearly stated/known that /sys/kernel/debug is required/mounted for bootrr?

File: Makefile (hunk 1 above)
Comment (aragua) [reply Makefile:14]: If #19 is merged I can rebase my patches above.

File: helpers/assert_file_is_empty (hunk 2 above)
Comment (aragua) [reply helpers/assert_file_is_empty]: sure I will change

File: helpers/bootrr-generic-tests (hunk 3 above)
Comment (aragua) [reply helpers/bootrr-generic-tests:3]: I ignore if /sys/kernel/debug is required. Someone knows?
If not, I can probably check it and skip test that depend on it.

Comment (eballetbo) [reply helpers/bootrr-generic-tests:3]: Yes, it is. So probably is worth check first if /sys/kernel/debug exists and if not just signal lava to skip the test?

Comment (aragua) [reply helpers/bootrr-generic-tests:3]: In the scope of assert_file_is_empty, I test first if the file exists and return skip if not

**Comment by aragua** [comment]
Any news or comments about this patch.
Thanks

**Pull request merged**